src/githubHelper.ts

```diff
@@ -160,6 +160,62 @@ export default class GithubHelper {
 
   // ----------------------------------------------------------------------------
 
+  
+  /**
+   * Gets a release by tag name
+   * @param tag {string} - the tag name to search a release for
+   * @returns 
+   */
+   async getReleaseByTag(tag) {
+    try {
+      await utils.sleep(this.delayInMs);
```

Comment:
Is the sleep here neccessary?

Reply:
it seems like this is a pattern we are following whenever we make the GitHub api call, to make sure that we do not exceed the quota. I have added this based on my understanding of the existing code

Reply:
For now we can leave it in. I just merged PRs relating to quatas / abuse limits. If it turns out it makes the migration slower than needed we will remove it later.